### PR TITLE
[CAMEL-17341, CAMEL-17342] BacklogDebugger API enhancements

### PR DESCRIPTION
**Pull request by javaduke**
This PR adds the following functionality to the BacklogDebugger MBean:
- Exchange Properties are now exposed via JMX as XML;
- An ability to evaluate an expression and return result.

### CODE REVIEW

File: core/camel-management/pom.xml [hunk 1]
```diff
@@ -85,6 +85,13 @@
             <artifactId>log4j-slf4j-impl</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (davsclaus) [inline core/camel-management/pom.xml:88]: We do NOT want this dependency

File: core/camel-management-api/pom.xml [hunk 2]
```diff
@@ -43,7 +43,10 @@
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
         </dependency>
-
+        <dependency>
```
Comment (davsclaus) [inline core/camel-management-api/pom.xml:46]: Remove this added dependency

**Review by davsclaus — CHANGES_REQUESTED** [review]
We also need unit test for changes into camel-core
And update the documentation at
https://github.com/apache/camel/blob/main/docs/user-manual/modules/ROOT/pages/backlog-debugger.adoc

File: core/camel-management/pom.xml (hunk 1 above)
Comment (javaduke) [reply core/camel-management/pom.xml:88]: I need this dependency to test if the result is Serializable. I understand your point about debugger being used by many different tools, perhaps we need another method which evaluates an expression and converts it to String?

File: core/camel-management/src/main/java/org/apache/camel/management/mbean/ManagedBacklogDebugger.java [hunk 3]
```diff
@@ -274,4 +283,77 @@ public long getFallbackTimeout() {
...
+
+    @Override
+    public Object evaluateExpressionAtBreakpoint(String id, String language, String expression, String resultType) {
+        Exchange suspendedExchange = null;
+        try {
+            Language lan = camelContext.resolveLanguage(language);
+            suspendedExchange = backlogDebugger.getSuspendedExchange(id);
+            if (suspendedExchange != null) {
+                Object result = null;
+                Class resultClass = Class.forName(resultType);
```
Comment (davsclaus) [inline core/camel-management/src/main/java/org/apache/camel/management/mbean/ManagedBacklogDebugger.java:335]: You cannot use this class loading, Camel has a ClassResolver API via CamelContext that must be used.
https://github.com/apache/camel/blob/main/core/camel-api/src/main/java/org/apache/camel/spi/ClassResolver.java

Comment (javaduke) [reply core/camel-management/src/main/java/org/apache/camel/management/mbean/ManagedBacklogDebugger.java]: thanks, fixed!

File: core/camel-management/pom.xml (hunk 1 above)
Comment (javaduke) [reply core/camel-management/pom.xml]: No problem, I actually figured out how to check if the object is serializable without using third party library, it's gone now.

**Comment by davsclaus** [comment]
We should not introduce a new operation for dumping the exchange properties - I would rather add a boolean option to the existing dump, where you can specify if the dump should be with exchange properties, eg

```
    @ManagedOperation(description = "Dumps the messages in xml format from the suspended breakpoint at the given node")
    String dumpTracedMessagesAsXml(String nodeId, boolean includeExchangeProperties);
```

**Comment by davsclaus** [comment]
We need a JIRA ticket about this, and the commit messages should link to the JIRA ticket, eg CAMEL-12345: bla bla

**Comment by javaduke** [comment]
> We should not introduce a new operation for dumping the exchange properties - I would rather add a boolean option to the existing dump, where you can specify if the dump should be with exchange properties

Yes, I thought about this, but two things stopped me - one is that exchange properties are technically not part of the message, they are part of the Exchange, and the second problem is that the XML generation logic is implemented in the DefaultBacklogTracerEventMessage and it doesn't have the Exchange instance. Please let me know if you have any good idea of how this should be implemented.

**Comment by javaduke** [comment]
> We need a JIRA ticket about this, and the commit messages should link to the JIRA ticket, eg [CAMEL-12345](https://issues.apache.org/jira/browse/CAMEL-12345): bla bla

I don't think I have access to the Jira, can you please create one for me?

**Comment by davsclaus** [comment]
> ented.

Yeah I noticed that too, but I think we can improve both at the same time: tracer and debugger by allows to capture exchange properties too. So you can say (true|false) to include exchange properties. If doing so, then yes. the root tag is different to be able to embed the exchange.

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
Oh we can also do as you did, just embed the `<exchangeProperties>` in the existing xml, as this only happens when you say (true) so its a new API and the old is as-is.

**Comment by javaduke** [comment]
I agree the tracer needs to be improved, but for now I just do some strings manipulation. I'll review the tracer later and see if I can pass the exchange and thus simplify the code.

**Pull request merged**